src/dav.js

```diff
@@ -0,0 +1,376 @@
+   * Either returns a string if the node only contains text, or returns an
+   * array of non-text subnodes.
+   *
+   * @param {Object} propNode node to parse
+   * @return {string|Array} text content as string or array of subnodes, excluding text nodes
+   */
+  _parsePropNode (propNode) {
+    var content = null
+    if (propNode.constructor === Object) {
+      if (Object.keys(propNode).length === 0) {
+        return ''
+      }
+      var subNodes = []
+      // Propnode can be any one of these
+      //         { 'oc:share-type': [ '0', '3' ] }
+      //         { 'oc:share-type': '3' }
+      //         { 'd:collection': {} }
+      for (var key in propNode) {
+        var node = propNode[key]
+        if (typeof node !== 'object') {
+          subNodes.push(node)
+          continue
+        }
+        if (Array.isArray(node)) {
+          for (var item of node) {
+            subNodes.push(item)
+          }
+          continue
+        }
+        var nsComponent = key.split(':')[0]
+        var localComponent = key.split(':')[1]
+        var nsValue = this.xmlNamespacesComponents[nsComponent]
+        subNodes.push('{' + nsValue + '}' + localComponent)
+      }
+      if (subNodes.length) {
+        content = subNodes
+      }
+    } else if (propNode) {
+      content = propNode
+    } else {
+      content = ''
+    }
+
+    return content
+  }
+
+  /**
+   * Parses a multi-status response body.
+   *
+   * @param {string} xmlBody
+   * @param {Array}
+   */
+  parseMultiStatus (xmlBody) {
+    const doc = parser.xml2js(xmlBody)
+
+    var responseIterator = doc['d:multistatus']['d:response'] || []
+    if (responseIterator.constructor !== Array) {
+      responseIterator = [responseIterator]
+    }
+    var result = []
+    responseIterator.forEach(responseNode => {
+      var response = {
+        href: null,
+        propStat: []
+      }
+
+      response.href = responseNode['d:href']
+
+      var propStatIterator = responseNode['d:propstat']
+
+      if (propStatIterator.constructor !== Array) {
+        propStatIterator = [propStatIterator]
+      }
+      propStatIterator.forEach(propStatNode => {
+        var propStat = {
+          status: propStatNode['d:status'],
+          properties: {}
+        }
+
+        var propIterator = propStatNode['d:prop']
+        if (propIterator.constructor !== Array) {
+          propIterator = [propIterator]
+        }
+        propIterator.forEach(propNode => {
+          for (var key in propNode) {
+            var content = this._parsePropNode(propNode[key])
+            var nsComponent = key.split(':')[0]
+            var localComponent = key.split(':')[1]
+            var nsValue = this.xmlNamespacesComponents[nsComponent]
+            propStat.properties['{' + nsValue + '}' + localComponent] = content
+          }
+        })
+        response.propStat.push(propStat)
+      })
+
+      result.push(response)
+    })
+    return result
+  }
+
+  parseClarkNotation (propertyName) {
```

Comment:
unit tests for this function would be useful to make sure the regex is correct and stays correct